template/server.js

```diff
@@ -1,106 +1,110 @@
+    app,
+    templatePath,
+    (bundle, options) => {
+      renderer = createRenderer(bundle, options);
+    }
+  )
 }
 
-application.use(favicon(resolve('./static/favicon.ico')));
-if(isProduction) {
-  application.use('/static', express.static(resolve('./dist/static')));
-}
+const serve = (path, cache) => express.static(resolve(path), {
+  maxAge: cache && isProd ? 1000 * 60 * 60 * 24 * 30 : 0,
+});
 
-// writes the head of index.html
-// then injects the initial state of the application
-// then writes the application as it's being rendered
-// finally writes the tail of index.html
-application.get('*', (request, response) => {
-  if (!renderer) {
-    return response.end('Compiling... Refresh in a moment!');
+app.use(compression({ threshold: 0  }));
+app.use(favicon(resolve('./public/favicon.ico')));
+app.use('/dist', serve('./dist', true));
+app.use('/public', serve('./public', true));
+app.use('/manifest.json', serve('./manifest.json', true));
+app.use('/service-worker.js', serve('./dist/service-worker.js'));
+
+app.use(microcache.cacheSeconds(1, req => useMicroCache && req.originalUrl));
+
+function render (req, res) {
+  const s = Date.now();
+
+  res.setHeader("Content-Type", "text/html");
+  res.setHeader("Server", serverInfo);
+
+  const handleError = err => {
+    if (err.url) {
+      res.redirect(err.url)
+    } else if (err.code === 404) {
+      res.status(404).send('404 | Page Not Found');
+    } else {
+      // Render Error Page or Redirect
+      res.status(500).send('500 | Internal Server Error');
+      console.error(`error during render : ${req.url}`);
+      console.error(err.stack);
+    }
   }
 
-  const timestamp = Date.now();
-  let context = { url: request.originalUrl };
-  const renderStream = renderer.renderToStream(context);
-  let firstChunk = true;
-
-  response.write(indexHTML.head);
-
-  renderStream.on('data', chunk => {
-    if(firstChunk) {
-      // embed initial store state
-      if(context.initialState) {
-        response.write(
-          `<script>
-             window.INITIAL_STATE = ${serialize(context.initialState, { isJSON: true })};
-           </script>`
-        );
-      }
-
-      firstChunk = false;
+  const context = {
+    title: 'Big Spaceship Vue Template',
+    url: req.url,
+  }
+  renderer.renderToString(context, (err, html) => {
+    if (err) {
+      return handleError(err);
+    }
+    res.send(html);
+    if (!isProd) {
+      console.log(`whole request: ${Date.now() - s}ms`);
     }
-
-    response.write(chunk);
-  });
-
-  renderStream.on('end', () => {
-    response.end(indexHTML.tail);
-    console.log(`Request completed in ${Date.now() - timestamp}ms`);
   });
+}
 
-  renderStream.on('error', error => { throw error });
+app.get('*', isProd ? render : (req, res) => {
+  readyPromise.then(() => render(req, res));
 });
 
 const port = process.env.PORT || 8080;
-application.listen(port, () => {
+app.listen(port, () => {
   const uri = `http://localhost:${port}`;
```

Comment:
We could set up our locals to run over https (spoofed) so we're removing that variable difference from environments.